### DISCUSSION TITLE
rabbitmq_io_write_ops_total and rabbitmq_io_read_ops_total

### DISCUSSION THREAD

**Discussion by lanzhiwang** [rabbitmq/rabbitmq-website#1858] [General]
In the rabbitmq 3.12.4 version, the quorum queues is used to produce and consume messages, and the plug-in is used to output monitoring indicators to Prometheus.

Among the monitoring indicators, `rabbitmq_io_write_ops_total` and `rabbitmq_io_read_ops_total` are always 0.
It looks like there are no disk IO operations.

In fact, data is written to the disk, and the size of the wal file is constantly growing. This can mean that the data is written to the disk.

```bash
$ ls -al ./
total 95236
drwxr-xr-x 3 polkitd input 107 Mar 28 17:41 .
drwxr-xr-x 3 polkitd input 71 Mar 28 17:11 ..
-rw-r--r-- 1 polkitd input 64000194 Mar 28 17:41 00000043.wal
-rw-r--r-- 1 polkitd input 23984472 Mar 28 17:41 00000044.wal
drwxr-xr-x 3 polkitd input 8192 Mar 28 17:41 2F_THRUFCAY7FKEVRD
-rw-r--r-- 1 polkitd input 5570 Mar 28 17:41 meta.dets
-rw-r--r-- 1 polkitd input 5910 Mar 28 17:18 names.dets

$ ls -al ./
total 95236
drwxr-xr-x 3 polkitd input 107 Mar 28 17:41 .
drwxr-xr-x 3 polkitd input 71 Mar 28 17:11 ..
-rw-r--r-- 1 polkitd input 64000194 Mar 28 17:41 00000043.wal
-rw-r--r-- 1 polkitd input 29078061 Mar 28 17:41 00000044.wal
drwxr-xr-x 3 polkitd input 8192 Mar 28 17:41 2F_THRUFCAY7FKEVRD
-rw-r--r-- 1 polkitd input 5570 Mar 28 17:41 meta.dets
-rw-r--r-- 1 polkitd input 5910 Mar 28 17:18 names.dets
```

I would like to ask why the above two monitoring indicators are always 0. Is it a problem with the monitoring plug-in itself or the mechanism of the quorum queues itself?

**Comment by Zerpet** [comment]
Those metrics are used in Classic Queues v1 and little else. Quorum Queues and Streams do not update IO metrics. You will get better IO metrics from the OS, for example, with [Prometheus Node Exporter](https://prometheus.io/docs/guides/node-exporter/). Those metrics are likely to be removed in RabbitMQ 4.0.
